lib/json/pure/generator.rb

```diff
@@ -354,7 +354,7 @@ def json_transform(state)
               result << delim unless first
               result << state.indent * depth if indent
               result = "#{result}#{key.to_s.to_json(state)}#{state.space_before}:#{state.space}"
-              if state.strict?
+              if state.strict? && !(!value || value == true || value.is_a?(Array) || value.is_a?(Integer) || value.is_a?(Float) || value.is_a?(String))
```

Comment:
This looks expensive, can't we just rely on `.to_json(state)` calling e.g. Object#to_json and that will raise?

Reply:
I don't think we can as it wouldn't catch objects with a custom `to_json`.

Reply:
Re `!value` that could be an issue if someone redefines `!` on their type to return true in some cases.
Not sure whether it's worth caring about, redefining `!` seems a pretty poor idea (maybe it should be a Ruby warning even).